fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -325,26 +328,26 @@ private CompletableFuture<Void> handleCrossPartitionDiscontinuity() {
         // with respect to the continuation, do that now:
         return partitioner.getPartitionMetaInfoById(Objects.requireNonNull(partitionId), groupingKey).thenCompose(partitionInfo -> {
             // the boundaries of the partition may have changed (due to re-balancing, explicit doc removal etc.)
-            long currentFrom = LucenePartitioner.getFrom(partitionInfo);
-            long currentTo = LucenePartitioner.getTo(partitionInfo);
-            partitionTimestamp = currentFrom;
+            Tuple currentFrom = Tuple.fromBytes(partitionInfo.getFrom().toByteArray());
+            Tuple currentTo = Tuple.fromBytes(partitionInfo.getTo().toByteArray());
+            partitionKey = currentFrom;
 
             final CompletableFuture<Void> getProperContinuationPartition;
 
-            // next, if the current partition's new boundary excludes the continuation's "search after" timestamp,
-            // we need to change the current partition to the one that does contain this timestamp.
-            if (searchAfterTimestamp != null && (searchAfterTimestamp < currentFrom || searchAfterTimestamp > currentTo)) {
+            // next, if the current partition's new boundary excludes the continuation's "search after" partitioning key,
+            // we need to change the current partition to the one that does contain this partitioning key.
+            if (searchAfterPartitioningKey != null && (searchAfterPartitioningKey.compareTo(currentFrom) < 0 || searchAfterPartitioningKey.compareTo(currentTo) > 0)) {
 
                 // need to preserve the current "searchAfter" doc across the partition change
                 dontResetSearchAfter = true;
 
                 // get the proper partition
-                getProperContinuationPartition = partitioner.findPartitionInfo(groupingKey, searchAfterTimestamp - 1).thenCompose(properPartitionInfo -> {
+                getProperContinuationPartition = partitioner.findPartitionInfo(groupingKey, searchAfterPartitioningKey).thenCompose(properPartitionInfo -> {
```

Comment:
Why was it `-1` before, but now it is equality?
Should it be using `LucenePartitioner.getNextOlderPartitionInfo` ?

Reply:
`findPartitionInfo` implementation changed (it does range scan on the `partitionKey` instead of timestamp, inclusive)

Reply:
I'm not sure I see how `findPartitonInfo` changed in a relevant way.
It used to add the `timestamp` to the metadata space, and now it adds the `timestamp,primaryKey`, but the scan itself is the same.